src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/DataSetsControllerTests.cs

```diff
@@ -714,440 +719,859 @@ public GetDataSetMetaTests(TestApplicationFactory testApp) : base(testApp)
+                var content = response.AssertOk<DataSetMetaViewModel>(useSystemJson: true);
 
-            var response = await GetDataSetMeta(dataSet.Id, dataSetVersion.Version);
+                Assert.NotNull(content);
 
-            response.AssertForbidden();
-        }
+                Assert.NotEmpty(content.Filters);
+                Assert.NotEmpty(content.Locations);
+                Assert.NotEmpty(content.Indicators);
+                Assert.NotEmpty(content.TimePeriods);
+            }
 
-        [Fact]
-        public async Task VersionExistsForOtherDataSet_Returns404()
-        {
-            DataSet dataSet1 = DataFixture
-                .DefaultDataSet()
-                .WithStatusPublished();
+            [Theory]
+            [InlineData(MetadataType.Filters)]
+            [InlineData(MetadataType.Locations)]
+            [InlineData(MetadataType.Indicators)]
+            [InlineData(MetadataType.TimePeriods)]
+            public async Task OneTypeSpecified_OnlyReturnsMetadataForSpecifiedType(MetadataType metadataType)
+            {
+                DataSet dataSet = DataFixture
+                    .DefaultDataSet()
+                    .WithStatusPublished();
+
+                await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+                DataSetVersion dataSetVersion = DataFixture
+                    .DefaultDataSetVersion(
+                        filters: 1,
+                        indicators: 1,
+                        locations: 1,
+                        timePeriods: 2)
+                    .WithStatusPublished()
+                    .WithDataSetId(dataSet.Id)
+                    .FinishWith(dsv => dataSet.LatestVersion = dsv);
+
+                await TestApp.AddTestData<PublicDataDbContext>(context =>
+                {
+                    context.DataSetVersions.Add(dataSetVersion);
+                    context.DataSets.Update(dataSet);
+                });
 
-            DataSet dataSet2 = DataFixture
-                .DefaultDataSet()
-                .WithStatusPublished();
+                var response = await GetDataSetMeta(
+                    dataSet.Id,
+                    types: [metadataType.ToString()]);
 
-            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.AddRange(dataSet1, dataSet2));
+                var content = response.AssertOk<DataSetMetaViewModel>(useSystemJson: true);
 
-            DataSetVersion dataSetVersion = DataFixture
-                .DefaultDataSetVersion(
-                    filters: 1,
-                    indicators: 1,
-                    locations: 1,
-                    timePeriods: 3)
-                .WithStatusPublished()
-                .WithDataSetId(dataSet1.Id);
+                Assert.NotNull(content);
 
-            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSetVersions.Add(dataSetVersion));
+                switch (metadataType)
```

Comment:
Bit of a nitpick really, but as mentioned in another comment about the type itself, it'd be good to reference everything as 'meta' instead of 'metadata' for brevity and consistency with everywhere else where we call it 'meta'.

Similar thing could apply elsewhere in these new tests